dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +66,65 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__isCertInstalled: function() {
+		var self = this;
+		if (window.location.protocol === "https:") {
+			self.client.ping("controller", "https").catch(function (args) {
+				var xhr = args[1];
+				if (xhr.status === 0) {
+					// We were not able to access the controller due to a network error (ssl, timeout)
+					// In order to understand what's happening, we have to switch to http.
+					self.dispatcher.handleAppEvent({
+						name: "CONTROLLER_UNAVAILABLE_FROM_HTTPS",
+						status: xhr.status
+					});
+					return;
+				}
+				// We got something else than 0 and it's an error. This results in SERVICE_UNAVAILABLE
+				self.dispatcher.handleAppEvent({
+					name: "SERVICE_UNAVAILABLE",
+					status: xhr.status
+				});
+			});
+		} else {
+			self.client.ping("controller", "https").catch(function (httpsArgs) {
+				var httpsXhr = httpsArgs[1];
+				if (httpsXhr.status === 0) {
+					// https is unavailable, let's see if http works
+					self.client.ping("controller", "http").catch(function (httpArgs) {
+						var httpXhr = httpArgs[1];
+						if (httpXhr.status === 0) {
+							// https is failing as well...service is unavailable
+							self.dispatcher.handleAppEvent({
```

Comment:
If you move the functions passed to `catch` and `then` into another method on `self` it should be fine. Try to avoid anything more than 5 levels of indentation, I'm more concerned with large blocks of nested code where it's hard to see the outer control statement(s) at a glance.